### PR TITLE
Feature - Title click listener

### PR DESCRIPTION
**Pull request by quentin41500**
- Provide title click listener option for MCV

@prolificinteractive/android-devs

### CODE REVIEW

File: library/src/main/java/com/prolificinteractive/materialcalendarview/MaterialCalendarView.java [hunk 1]
```diff
@@ -1358,6 +1357,24 @@ public void setOnRangeSelectedListener(OnRangeSelectedListener listener) {
     }
 
     /**
+     * Add listener to the title.
+     *
+     * @param listener thing to be notified
+     */
+    public void setOnTitleClickListener(final OnClickListener listener) {
+        if (listener != null) {
+            title.setOnClickListener(listener);
```
Comment (tiensi) [inline library/src/main/java/com/prolificinteractive/materialcalendarview/MaterialCalendarView.java]: What's wrong with setting the listener to null if it should already be guarding for null listeners

File: sample/src/main/java/com/prolificinteractive/materialcalendarview/sample/DynamicSettersActivity.java [hunk 2]
```diff
@@ -43,6 +44,14 @@ protected void onCreate(Bundle savedInstanceState) {
         currentTileSize = MaterialCalendarView.DEFAULT_TILE_SIZE_DP;
         currentTileWidth = MaterialCalendarView.DEFAULT_TILE_SIZE_DP;
         currentTileHeight = MaterialCalendarView.DEFAULT_TILE_SIZE_DP;
+
+        widget.setOnTitleClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(final View view) {
+                Toast.makeText(DynamicSettersActivity.this, "Today is the day", Toast.LENGTH_SHORT)
```
Comment (tiensi) [inline sample/src/main/java/com/prolificinteractive/materialcalendarview/sample/DynamicSettersActivity.java]: string resources? or does it not matter

File: library/src/main/java/com/prolificinteractive/materialcalendarview/MaterialCalendarView.java [hunk 3]
```diff
@@ -222,7 +222,7 @@ public void onPageScrolled(int position, float positionOffset, int positionOffse
     private OnDateSelectedListener listener;
     private OnMonthChangedListener monthListener;
     private OnRangeSelectedListener rangeListener;
-
+    private OnClickListener titleListener;
```
Comment (tiensi) [inline library/src/main/java/com/prolificinteractive/materialcalendarview/MaterialCalendarView.java]: is this being used?

File: library/src/main/java/com/prolificinteractive/materialcalendarview/MaterialCalendarView.java [hunk 4]
```diff
@@ -1358,6 +1357,17 @@ public void setOnRangeSelectedListener(OnRangeSelectedListener listener) {
     }
 
     /**
+     * Add listener to the title or null to remove it.
+     *
+     * @param listener Listener to be notified.
+     */
+    public void setOnTitleClickListener(final OnClickListener listener) {
+        if (listener != null) {
+            title.setOnClickListener(listener);
```
Comment (tiensi) [inline library/src/main/java/com/prolificinteractive/materialcalendarview/MaterialCalendarView.java]: is there something wrong with setting anull click listener?

File: sample/src/main/java/com/prolificinteractive/materialcalendarview/sample/DynamicSettersActivity.java (hunk 2 above)
Comment (quentin41500) [inline sample/src/main/java/com/prolificinteractive/materialcalendarview/sample/DynamicSettersActivity.java]: 👍

File: library/src/main/java/com/prolificinteractive/materialcalendarview/MaterialCalendarView.java (hunk 3 above)
Comment (quentin41500) [inline library/src/main/java/com/prolificinteractive/materialcalendarview/MaterialCalendarView.java]: Nope

File: library/src/main/java/com/prolificinteractive/materialcalendarview/MaterialCalendarView.java (hunk 4 above)
Comment (quentin41500) [inline library/src/main/java/com/prolificinteractive/materialcalendarview/MaterialCalendarView.java]: Actually no, that's how you remove it! I will cleanup, thanks

**Comment by quentin41500** [comment]
@tiensi Updated following comments

**Pull request merged**